### ISSUE TITLE
ftx fetch_positions(symbols= ) doesn't return result for the specific symbol

### ISSUE THREAD

**Issue by kennethyuenwc** [ccxt/ccxt#10697]
When I run the code ftx fetch_positions(symbols='ETH-PERP), it returned all position I had traded, including those with no current position, instead of returning the opened position of the specific symbols.

Not seeing others having the same problem so i try to upgrade to the latest version of ccxt, but still not working.

- OS: Windows
- Programming Language version: Python 3.8.8
- CCXT version: 1.62.15

```
exchange.fetch_positions(symbols='ETH-PERP')
```

```
[{'info': {'future': 'BTC-PERP',
   'size': '0.0',
   'side': 'buy',
   'netSize': '0.0',
   'longOrderSize': '0.0',
   'shortOrderSize': '0.0',
   'cost': '0.0',
   'entryPrice': None,
   'unrealizedPnl': '0.0',
   'realizedPnl': '43.31487525',
   'initialMarginRequirement': '0.1',
   'maintenanceMarginRequirement': '0.03',
   'openSize': '0.0',
   'collateralUsed': '0.0',
   'estimatedLiquidationPrice': None,
   'recentAverageOpenPrice': None,
   'recentPnl': None,
   'recentBreakEvenPrice': None,
   'cumulativeBuySize': None,
   'cumulativeSellSize': None},
  'symbol': 'BTC-PERP',
  'timestamp': None,
  'datetime': None,
  'initialMargin': None,
  'initialMarginPercentage': 0.1,
  'maintenanceMargin': None,
  'maintenanceMarginPercentage': 0.03,
  'entryPrice': None,
  'notional': None,
  'leverage': 10,
  'unrealizedPnl': None,
  'contracts': 0.0,
  'contractSize': 1.0,
  'marginRatio': None,
  'liquidationPrice': None,
  'markPrice': None,
  'collateral': None,
  'marginType': 'cross',
  'side': 'long',
  'percentage': None},
 {'info': {'future': 'ETH-PERP',
   'size': '1.81',
   'side': 'sell',
   'netSize': '-1.81',
   'longOrderSize': '0.0',
   'shortOrderSize': '0.0',
   'cost': '-7655.033',
   'entryPrice': '4229.3',
   'unrealizedPnl': '0.0',
   'realizedPnl': '-615.26048801',
   'initialMarginRequirement': '0.1',
   'maintenanceMarginRequirement': '0.03',
   'openSize': '1.81',
   'collateralUsed': '765.5033',
   'estimatedLiquidationPrice': '13756.0814500442',
   'recentAverageOpenPrice': '4265.055248618784',
   'recentPnl': '64.717',
   'recentBreakEvenPrice': '4265.055248618784',
   'cumulativeBuySize': '0.0',
   'cumulativeSellSize': '1.81'},
  'symbol': 'ETH-PERP',
  'timestamp': None,
  'datetime': None,
  'initialMargin': 765.5033,
  'initialMarginPercentage': 0.1,
  'maintenanceMargin': 229.65099,
  'maintenanceMarginPercentage': 0.03,
  'entryPrice': None,
  'notional': 7655.033,
  'leverage': 10,
  'unrealizedPnl': 64.717,
  'contracts': 1.81,
  'contractSize': 1.0,
  'marginRatio': 0.0131,
  'liquidationPrice': 13756.0814500442,
  'markPrice': 4229.3,
  'collateral': 17473.12541458,
  'marginType': 'cross',
  'side': 'short',
  'percentage': 8.45}]
```

**Comment by frosty00** [comment]
@kennethyuenwc symbols must be a list of symbols

**Comment by kroitor** [comment]
```Python
exchange.fetch_positions(symbols=['ETH-PERP'])
```

**Comment by kennethyuenwc** [comment]
thanks for the quick response, but it still return the same output even I changed the symbols input to a list with exchange.fetch_positions(symbols=['ETH-PERP'])

**Comment by kroitor** [comment]
@kennethyuenwc looks like the filtering is missing there, will fix soon, thx for pointing it out!

**Comment by kroitor** [comment]
@kennethyuenwc pushed a fix to it, please, upgrade to CCXT 1.62.21+ (the new build will arrive in ~20 minutes) and let us know if that solves the issue for you or not. Thanks again!

**Comment by kennethyuenwc** [comment]
@kroitor thanks, just updated and tried again, now exchange.fetch_positions(symbols=['ETH-PERP']) return an empty list, but I do have positions for ETH-PERP

**Comment by kroitor** [comment]
@kennethyuenwc if possible, please, paste your verbose output:

```Python
exchange.verbose = True
positions = exchange.fetch_positions(symbols=['ETH-PERP'])
```

**Comment by kennethyuenwc** [comment]
Please refer to below,


```
Request: GET https://ftx.com/api/positions?showAvgPrice=true {'FTX-KEY': '   ', 'FTX-TS': '   ', 'FTX-SIGN': '   ', 'FTX-SUBACCOUNT': '   ', 'User-Agent': 'python-requests/2.25.1', 'Accept-Encoding': 'gzip, deflate'} None

Response: GET https://ftx.com/api/positions?showAvgPrice=true 200 {'Date': 'Fri, 26 Nov 2021 02:15:19 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Vary': 'Accept-Encoding, Accept-Encoding, Origin', 'Cache-Control': 'private, max-age=0, no-cache', 'account-id': '   ', 'ETag': 'W/"   "', 'Content-Encoding': 'gzip', 'X-Cache-Status': 'MISS', 'CF-Cache-Status': 'MISS', 'Expect-CT': 'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains; preload', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '   -HKG', 'alt-svc': 'h3=":443"; ma=86400, h3-29=":443"; ma=86400, h3-28=":443"; ma=86400, h3-27=":443"; ma=86400'} {"success":true,"result":[{"future":"BTC-PERP","size":0.524,"side":"buy","netSize":0.524,"longOrderSize":0.0,"shortOrderSize":0.0,"cost":30749.368,"entryPrice":58682.0,"unrealizedPnl":0.0,"realizedPnl":-262.85358758,"initialMarginRequirement":0.1,"maintenanceMarginRequirement":0.03,"openSize":0.524,"collateralUsed":3074.9368,"estimatedLiquidationPrice":28844.653927156487,"recentAverageOpenPrice":59262.75381679389,"recentPnl":-304.315,"recentBreakEvenPrice":59262.75381679389,"cumulativeBuySize":0.524,"cumulativeSellSize":0.0},{"future":"ETH-PERP","size":7.26,"side":"buy","netSize":7.26,"longOrderSize":0.0,"shortOrderSize":0.0,"cost":32652.576,"entryPrice":4497.6,"unrealizedPnl":0.0,"realizedPnl":-245.38977759,"initialMarginRequirement":0.1,"maintenanceMarginRequirement":0.03,"openSize":7.26,"collateralUsed":3265.2576,"estimatedLiquidationPrice":2344.0505038333336,"recentAverageOpenPrice":4375.178925619834,"recentPnl":888.777,"recentBreakEvenPrice":4375.178925619834,"cumulativeBuySize":7.26,"cumulativeSellSize":0.0}]}
[]
```

**Comment by kroitor** [comment]
@kennethyuenwc from your verbose response it looks like you don't have an open ETH-PERP position?

**Comment by kennethyuenwc** [comment]
I do, it can be seen from the extract of the verbose response
```
...
{"future":"ETH-PERP","size":7.26,"side":"buy","netSize":7.26
...
```

**Comment by kroitor** [comment]
@kennethyuenwc pardon, my bad, missed it, debugging now.

**Comment by kroitor** [comment]
@kennethyuenwc finally, found the bug, pushed a fix to it in CCXT 1.62.31. The new build will arrive in ~20 minutes as always. Tested on my side – should work smoothly now. Please, let me know if you still have difficulties with after upgrading. Your feedback is very much appreciated.

**Comment by kennethyuenwc** [comment]
@kroitor, It's working well finally, thanks for the hard work!
